### PR TITLE
test_node: get_mem_rss fixups

### PR DESCRIPTION
**Pull request by maflcko**
Follow up to #14522:

* Fix math (Memory usage increase relative to previous memory usage, not final memory usage)
* remove `shell=True`
* assert that the node is running
* Make it work on BSD-like systems

### CODE REVIEW

File: test/functional/test_framework/test_node.py [hunk 1]
```diff
@@ -118,22 +118,20 @@ def get_deterministic_priv_key(self):
...
+                stderr=subprocess.DEVNULL).strip())
 
-        # Catching `Exception` broadly to avoid failing on platforms where ps
-        # isn't installed or doesn't work as expected, e.g. OpenBSD.
+        # Catching to avoid failing on platforms where ps
+        # isn't installed (SubprocessError) or doesn't work as expected (ValueError), e.g. BSD.
         #
         # We could later use something like `psutils` to work across platforms.
-        except Exception:
+        except (subprocess.SubprocessError, ValueError):
```
Comment (jamesob) [inline test/functional/test_framework/test_node.py]: We could also try detecting/parsing the *BSD-style output, which looks like `"RSS\n 123"`.

**Review by jamesob — APPROVED** [review]
ACK 

Thanks for the fixes!

**Comment by lucash-dev** [comment]
utACK

**Comment by practicalswift** [comment]
Concept ACK

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/14693/commits/fa9ed38d57042c8a69699903a546c7bc0c10aaff

**Pull request merged**